crypto/fipsmodule/ec/make_tables.go

```diff
@@ -341,6 +412,35 @@ func bigIntToU64s(curve elliptic.Curve, n *big.Int) []uint64 {
 	return ret
 }
 
+// Convert big int to an array of 58-bit digits.
+// This is needed for P-521 Fiat-crypto implementation.
+func bigIntToU58s(curve elliptic.Curve, n *big.Int) []uint64 {
+	words := (curve.Params().BitSize + 57) / 58
+	ret := make([]uint64, words)
+	mask := big.NewInt((1 << 58) - 1)
+	tmp := new(big.Int).Set(n)
+	for i := 0; i < words; i++ {
+		ret[i] = new(big.Int).And(tmp, mask).Uint64()
+		tmp.Rsh(tmp, 58)
+	}
+	return ret
+}
+
+// Convert big int to an array of digits where each digit
+// has bit-size as specified in the input bitSizes array
+// This is needed for P-521 Fiat-crypto implementation.
+func bigIntToUCustom(curve elliptic.Curve, n *big.Int, bitSizes []uint) []uint64 {
+	words := len(bitSizes)
+	ret := make([]uint64, words)
+	tmp := new(big.Int).Set(n)
+  for i, bits := range bitSizes {
+    mask := big.NewInt((1 << bits) - 1)
+		ret[i] = new(big.Int).And(tmp, mask).Uint64()
+		tmp.Rsh(tmp, bits)
+	}
```

Comment:
Similar comment, not sure if there is a value in checking tmp=0 here.
Also the indentation seems off in the function.

Reply:
see my comment above. Indentation fixed.